From fedb7c4e28361164a7ae285c74b9a9590825bfbc Mon Sep 17 00:00:00 2001
From: "jie.yuan" <jie.yuan@amlogic.com>
Date: Wed, 13 Mar 2019 15:09:00 +0800
Subject: [PATCH] camera : set mirror flag to ensure preview correctly for usb
 camera [1/1]

PD#TV-3119

Problem:
Left and right positions are opposite

Solution:
set mirror flag to ensure preview correctly for usb camera

Verify:
verify it on ampere

Change-Id: Ib1acb32b1800994400f02fd62a3eba5120df21c6
Signed-off-by: jie.yuan <jie.yuan@amlogic.com>
---
 camera/CameraUtils.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/camera/CameraUtils.cpp b/camera/CameraUtils.cpp
index 67fc116..37df884 100644
--- a/camera/CameraUtils.cpp
+++ b/camera/CameraUtils.cpp
@@ -52,7 +52,8 @@ status_t CameraUtils::getRotationTransform(const CameraMetadata& staticInfo,
 
     int32_t& flags = *transform;
 
-    bool mirror = (entryFacing.data.u8[0] == ANDROID_LENS_FACING_FRONT);
+    bool mirror = (entryFacing.data.u8[0] == ANDROID_LENS_FACING_FRONT)
+           || (entryFacing.data.u8[0] == ANDROID_LENS_FACING_EXTERNAL);
     int orientation = entry.data.i32[0];
     if (!mirror) {
         switch (orientation) {
-- 
2.7.4

